source/server/http/admin.cc

```diff
@@ -622,18 +630,24 @@ Http::Code AdminImpl::runCallback(const std::string& path_and_query,
   return code;
 }
 
+std::vector<const AdminImpl::UrlHandler*> AdminImpl::sortedHandlers() const {
+  std::vector<const UrlHandler*> sorted_handlers;
+  for (const UrlHandler& handler : handlers_) {
+    sorted_handlers.push_back(&handler);
+  }
+  // Note: it's generally faster to sort a vector with std::vector than to construct a std::map.
```

Comment:
I agree. One other consideration is that it might be preferable to have less optimal but more readable/shorter code unless on the data plane fast path.

Reply:
Fair enough, although I don't think the vector-population & sorting code is that hard to read.  It would be shorter too if I could figure out how to directly convert a list of structs into a vector of pointers to structs in STL; seems like that should be possible without writing a loop but I couldn't find the magic algorithm for that.

Reply:
I'm fine either way.